### DISCUSSION TITLE
.pm is detecting as raku

### DISCUSSION THREAD

**Discussion by ricks03** [github-linguist/linguist#5371] [Classification] [answered]
The .pm files in my repo are being detected as raku. While .pm might indeed be a Perl 6 extension, it's also a Perl 5 extension.

**Comment by lildude** [comment, marked as answer]
Indeed, and it is already associated with Perl:

https://github.com/github/linguist/blob/19b3a9bd7d585be8d6459cfab0a51607b2aee68b/lib/linguist/languages.yml#L4203-L4220

If the file is being classed as Raku in your repo it means it either matches the heuristic:

https://github.com/github/linguist/blob/19b3a9bd7d585be8d6459cfab0a51607b2aee68b/lib/linguist/heuristics.yml#L397-L402

https://github.com/github/linguist/blob/19b3a9bd7d585be8d6459cfab0a51607b2aee68b/lib/linguist/heuristics.yml#L638-L639

… or it has been guessed by the classifier which will make the  guess based on the samples Linguist has. This won't be perfect so you may need to implement an override. We'd welcome a PR if you think the heuristic needs improving.
